src/Microsoft.OData.Client/DataServiceClientFormat.cs

```diff
@@ -175,12 +175,14 @@ internal void SetRequestAcceptHeaderForCount(HeaderCollection headers)
         }
 
         /// <summary>
-        /// Sets the value of the Accept header for a count request (will set it to 'multipart/mixed').
+        /// Sets the value of the Accept header for a batch request
+        /// Will set it to 'multipart/mixed' for a multipart batch request
+        /// Will set it to 'application/json;odata.metadata=minimal' for a json batch request
         /// </summary>
         /// <param name="headers">The headers to modify.</param>
-        internal void SetRequestAcceptHeaderForBatch(HeaderCollection headers)
+        internal void SetRequestAcceptHeaderForBatch(HeaderCollection headers, bool useJsonBatch)

```

Comment:
Do you really need this bool param? It's possible to infer if it's json based on the headers collection

Reply:
I found this abit straight forward

Reply:
It seems unnecessary to modify the method signature since inferring within the method is also quite straightforward

Reply:
I have fixed this.